include/deal.II/base/symmetric_tensor.h

```diff
@@ -3544,6 +3544,121 @@ DEAL_II_HOST constexpr inline DEAL_II_ALWAYS_INLINE
   return result;
 }
 
+/**
+ * An enumeration to describe the desired symmetry of the
+ * symmetric fourth order tensor
+ */
+enum class SymmetryType
```

Comment:
Would you be willing to rename this to `TensorSymmetryType` to make it more specific to just tensors? Symmetry is a concept that appears in many places, and I'm concerned about using an overly generic name.